File: src/main/java/net/guneyilmaz0/mongos/MongoS.java
package net.guneyilmaz0.mongos;

import com.mongodb.MongoClient;
import com.mongodb.MongoClientURI;
import com.mongodb.client.MongoCollection;
import lombok.Getter;
import org.bson.Document;

@Getter
@SuppressWarnings("unused")
public class MongoS extends Database  {
    private final MongoClient mongo;

    public MongoS(String host, int port, String dbName) {
        this.mongo = new MongoClient(host, port);
        init(mongo.getDatabase(dbName));
    }

    public MongoS(MongoClientURI uri, String dbName) {
        this.mongo = new MongoClient(uri);
        init(mongo.getDatabase(dbName));
    }

    public MongoS(String dbName){
        this.mongo = new MongoClient("localhost", 27017);
        init(mongo.getDatabase(dbName));
    }

    public MongoCollection<Document> getCollection(String collection) {
        return this.database.getCollection(collection);
    }

    public Database getAnotherDatabase(String dataBase) {
        Database db = new Database();
        db.init(mongo.getDatabase(dataBase));
        return db;
    }
}

File: src/main/kotlin/net/guneyilmaz0/mongos/MongoS.kt
package net.guneyilmaz0.mongos

import com.mongodb.client.MongoClient
import com.mongodb.client.MongoClients
import com.mongodb.client.MongoCollection
import com.mongodb.client.ChangeStreamIterable
import org.bson.Document

@Suppress("unused", "MemberVisibilityCanBePrivate")
class MongoS : Database {
    private val mongo: MongoClient

    constructor(host: String, port: Int, dbName: String) {
        mongo = MongoClients.create("mongodb://$host:$port")
        init(mongo.getDatabase(dbName))
    }

    constructor(uri: String, dbName: String) {
        mongo = MongoClients.create(uri)
        init(mongo.getDatabase(dbName))
    }

    constructor(dbName: String) {
        mongo = MongoClients.create()
        init(mongo.getDatabase(dbName))
    }

    fun getCollection(collection: String): MongoCollection<Document> = database!!.getCollection(collection)

    fun getAnotherDatabase(dataBase: String): Database {
        val db = Database()
        db.init(mongo.getDatabase(dataBase))
        return db
    }

    fun watchCollection(collection: String): ChangeStreamIterable<Document> = getCollection(collection).watch()

}